### PR TITLE
Add baseURL build script

### PR DESCRIPTION
**Pull request by kaladin9017**
#### What's this PR do?

Added the script to generate a new hugo config file before every hugo build using the BASE_URL env variable from netlify.

#### How was this tested? How should this be reviewed?
I tested it on my local build process and using Hugo server. 

#### What are the relevant tickets?
Relevant Github issues

#### Questions / Considerations

The other changed files are for adding links to categories and authors from article-preview cards and featured article display.

### CODE REVIEW

File: src/less/blog-v2/includes/latest-articles.less [hunk 1]
```diff
@@ -21,6 +21,12 @@
     padding: @category-padding;
     background-color: @color-category-card-tag-background;
     color: @color-category-card-tag-font;
+    a {
```
Comment (jonuy) [inline src/less/blog-v2/includes/latest-articles.less]: The spacing got kinda wonky in the .less files. Is this something that Prettier did?

File: package.json [hunk 2]
```diff
@@ -3,13 +3,15 @@
   "version": "1.0.0",
   "description": "Shine blog framework",
   "scripts": {
+    "add-baseurl":"node hugo/lib/getUrl.js",
     "build": "npm run webpack-build && npm run hugo-build",
-    "build-for-staging": "npm run clear-public && npm run clear-content && npm run fetch-content -- --staging && npm run build",
+    "build-for-staging": "npm run clear-public && npm run clear-content && npm run add-baseurl && npm run fetch-content -- --staging && npm run build",
     "build-for-production": "npm run clear-public && npm run clear-content && npm run fetch-content -- --production && npm run build",
```
Comment (jonuy) [inline package.json]: Do you think we should add `npm run add-baseurl` to the `build-for-production` command too?

File: package.json [hunk 3]
```diff
@@ -3,13 +3,15 @@
...
     "clear-content": "cd hugo/lib && ./clear-content.sh",
     "clear-public": "rm -rf ./public",
     "fetch-content": "node hugo/lib/getArticles.js",
-    "hugo-build": "hugo --config=config/hugo.config.toml --destination=public/",
+    "hugo-build": "npm run add-baseurl &&hugo --config=config/hugo.config.toml --destination=public/",
```
Comment (jonuy) [inline package.json]: 👾  This is minor, but just for the sake of readability we can add a space between `&&` and `hugo`

Comment (kaladin9017) [inline package.json]: Copy. Missed that >_<.

File: package.json (hunk 2 above)
Comment (kaladin9017) [inline package.json]: I was going to make sure everything runs well on staging before I added it to production script but I can add it now.

File: src/less/blog-v2/includes/latest-articles.less (hunk 1 above)
Comment (kaladin9017) [inline src/less/blog-v2/includes/latest-articles.less]: Yep.

**Comment by kaladin9017** [comment]
It felt wrong as I was doing it but I am a bit hesitant to try git techniques I'm not familiar with. Will look into ``` git commit --amend ```, and find some time to dive deeper into version control.

**Pull request merged**